src/main/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringcreateanorderapi/resource/DeviceWearerContactDetailsController.kt

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.resource
+
+import jakarta.persistence.EntityNotFoundException
+import jakarta.validation.ConstraintViolationException
+import jakarta.validation.Valid
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.Authentication
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.models.DeviceWearerContactDetails
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.models.ErrorMessage
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.resource.validator.ValidationError
+import uk.gov.justice.digital.hmpps.hmppselectronicmonitoringcreateanorderapi.service.DeviceWearerContactDetailsService
+import java.util.*
+
+@RestController
+@PreAuthorize("hasRole('ROLE_EM_CEMO__CREATE_ORDER')")
+@RequestMapping("/api/")
+class DeviceWearerContactDetailsController(
+  @Autowired val contactDetailsService: DeviceWearerContactDetailsService,
+) {
+
+  @ExceptionHandler(EntityNotFoundException::class)
```

Comment:
You don't need this here, just throw NoResourceFoundException and it would handled by [HmppsElectronicMonitoringCreateAnOrderApiExceptionHandler](https://github.com/ministryofjustice/hmpps-electronic-monitoring-create-an-order-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringcreateanorderapi/config/HmppsElectronicMonitoringCreateAnOrderApiExceptionHandler.kt)

Reply:
You can add handler to HmppsElectronicMonitoringCreateAnOrderApiExceptionHandler to handle EntityNotFoundException. You don't need the handler added to every controller

Reply:
Same for ConstraintViolationException